hazelcast/src/main/java/com/hazelcast/map/impl/query/MapQueryEngineImpl.java

```diff
@@ -275,6 +280,11 @@ public QueryResult newQueryResult(int numberOfPartitions) {
         return new QueryResult(queryResultSizeLimiter.getNodeResultLimit(numberOfPartitions));
     }
 
+    @Override
+    public Predicate optimize(Predicate predicate, Indexes indexes) {
```

Comment:
Should this not be an implementation detail? Now you expose the fact that queries can be optimized. It is not relevant to the outside that this is in place.